File: backend/src/main/java/com/project/entities/AssessmentQuestion.java
package com.project.entities;

import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
@Table(name = "assessment_questions")
public class AssessmentQuestion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long questionId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assessment_id", nullable = false)
    private Assessment assessment;

    private String questionText;
    private Integer orderIndex;

    @OneToMany(mappedBy = "question", fetch = FetchType.LAZY)
    private List<AssessmentOption> options;
}

File: backend_dotnet/SafeMind.Core/Entities/AssessmentQuestion.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SafeMind.Core.Entities
{
    [Table("assessment_questions")]
    public class AssessmentQuestion
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("question_id")]
        public long QuestionId { get; set; }

        [Required]
        [ForeignKey(nameof(Assessment))]
        [Column("assessment_id")]
        public long AssessmentId { get; set; }

        public Assessment Assessment { get; set; } = null!;

        [Column("question_text")]
        public string? QuestionText { get; set; }
        [Column("order_index")]
        public int? OrderIndex { get; set; }

        public ICollection<AssessmentOption> Options { get; set; } = new List<AssessmentOption>();
    }
}
